extensions/cl_intel_device_attribute_query.asciidoc

```diff
@@ -0,0 +1,197 @@
+#define CL_DEVICE_NUM_THREADS_PER_EU_INTEL        0x4255
+#define CL_DEVICE_FEATURE_CAPABILITIES_INTEL      0x4256
+----
+
+Bitfield type describing the feature capabilities of a device.
+Additional feature flags may be added in subsequent versions of the extension.
+
+[source]
+----
+typedef cl_bitfield         cl_device_feature_capabilities_intel;
+
+/* For GPU devices, version 1.0.0: */
+
+#define CL_DEVICE_FEATURE_FLAG_DP4A_INTEL         (1 << 0)
+#define CL_DEVICE_FEATURE_FLAG_DPAS_INTEL         (1 << 1)
+----
+
+== New API Types
+
+None.
+
+== Modifications to the OpenCL API Specification
+
+(Add to the list preceding Table 5 in Section 4.2 - Querying Devices) ::
++
+--
+The device queries described in the Device Queries table should return the same information for a root-level device i.e. a device returned by *clGetDeviceIDs* and any sub-devices created from this device except for the following queries:
+
+* ...
+* {CL_DEVICE_NUM_SLICES_INTEL}
+--
+
+(Add to Table 5 - OpenCL Device Queries in Section 4.2 - Querying Devices) ::
++
+--
+[caption="Table 5. "]
+.List of supported param_names by clGetDeviceInfo
+[width="100%",cols="<30%,<20%,<50%",options="header"]
+|====
+| *cl_device_info* | Return Type | Description
+| {CL_DEVICE_IP_VERSION_INTEL}
+  | `cl_version`
+      | The IP version for the device.
+        The meaning of this version is implementation-defined, but newer devices should have a higher version than older devices.
+| {CL_DEVICE_ID_INTEL}
+  | `cl_uint`
+      | A unique device identifier for the OpenCL device.
+
+        If the implementation is driven primarily by a PCI device with a PCI device ID, the low 16 bits of the device identifier must contain that PCI device ID, and the remaining bits must be set to zero.
+        Otherwise, the choice of what to return may be dictated by operating system or platform policies - but should uniquely identify both the device version and any major configuration options (for example, core count in the case of multi-core devices).
+| {CL_DEVICE_NUM_SLICES_INTEL}
+  | `cl_uint`
+      | The number of slices in the device.
+        A slice is a collection of sub-slices.
+| {CL_DEVICE_NUM_SUB_SLICES_PER_SLICE_INTEL}
+  | `cl_uint`
+      | The maximum number of sub-slices in one slice.
+        A slice is a collection of execution units (EUs), fixed-function units, and caches.
+| {CL_DEVICE_NUM_EUS_PER_SUB_SLICE_INTEL}
+  | `cl_uint`
+      | The maximum number of execution units (EUs) in one sub-slice.
+        An execution unit is a multi-threaded processor that runs OpenCL kernels.
+| {CL_DEVICE_NUM_THREADS_PER_EU_INTEL}
+  | `cl_uint`
+      | The maximum number of threads that may be simultaneously resident on one execution unit (EU).
+| {CL_DEVICE_FEATURE_CAPABILITIES_INTEL}
+  | {cl_device_feature_capabilities_intel_TYPE}
+      | Feature flags describing capabilities supported by the device.
+|====
+--
+
+== Issues
+
+. How to express cache sizes?  Is for the entire device?
```

Comment:
Second sentence could use an edit. Maybe refer to the scope of the cache(s) being queried?

Reply:
Yes, it should be:

```suggestion
. How to express cache sizes?  Is it for the entire device?
```

Thanks!